sql/core/src/main/java/org/apache/spark/sql/sources/v2/MicroBatchReadSupport.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.spark.sql.sources.v2;
+
+import java.util.Optional;
+
+import org.apache.spark.annotation.InterfaceStability;
+import org.apache.spark.sql.sources.v2.reader.MicroBatchReader;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * A mix-in interface for {@link DataSourceV2}. Data sources can implement this interface to
+ * provide streaming micro-batch data reading ability.
+ */
+@InterfaceStability.Evolving
+public interface MicroBatchReadSupport extends DataSourceV2 {
+  /**
+   * Creates a {@link MicroBatchReader} to scan a batch of data from this data source.
```

Comment:
nit: MicrobatchReader does not "a batch". How about "to read batches of data from this data source in a streaming query"

Reply:
As spoken offline, add more docs specifying the lifecycle of the reader. creation -> [setOffsetRange -> createReadTasks]* -> stop